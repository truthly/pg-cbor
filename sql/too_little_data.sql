BEGIN;

CREATE EXTENSION IF NOT EXISTS cbor;

SELECT cbor.to_jsonb('\x4cf09fa7acf09f909863626f72');

SAVEPOINT too_little_data;
SELECT cbor.to_jsonb('\x4cf09fa7acf09f909863626f');
ROLLBACK TO too_little_data;

SELECT cbor.to_jsonb('\x6cf09fa7acf09f909863626f72');

SAVEPOINT too_little_data;
SELECT cbor.to_jsonb('\x6cf09fa7acf09f909863626f');
ROLLBACK TO too_little_data;

ROLLBACK;
