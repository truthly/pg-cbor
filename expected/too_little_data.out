BEGIN;
CREATE EXTENSION IF NOT EXISTS cbor;
SELECT cbor.to_jsonb('\x4cf09fa7acf09f909863626f72');
          to_jsonb          
----------------------------
 "f09fa7acf09f909863626f72"
(1 row)

SAVEPOINT too_little_data;
SELECT cbor.to_jsonb('\x4cf09fa7acf09f909863626f');
ERROR:  too little data
CONTEXT:  PL/pgSQL function cbor.major_type_2(bytea,text,integer,integer,numeric) line 5 at RAISE
PL/pgSQL function cbor.next_item(bytea,text) line 23 at RETURN
SQL function "to_jsonb" statement 1
ROLLBACK TO too_little_data;
SELECT cbor.to_jsonb('\x6cf09fa7acf09f909863626f72');
 to_jsonb 
----------
 "üß¨üêòcbor"
(1 row)

SAVEPOINT too_little_data;
SELECT cbor.to_jsonb('\x6cf09fa7acf09f909863626f');
ERROR:  too little data
CONTEXT:  PL/pgSQL function cbor.major_type_3(bytea,text,integer,integer,numeric) line 5 at RAISE
PL/pgSQL function cbor.next_item(bytea,text) line 20 at RETURN
SQL function "to_jsonb" statement 1
ROLLBACK TO too_little_data;
ROLLBACK;
